packages/fxa-settings/src/components/Settings/FlowSetup2faPrompt/index.tsx

```diff
@@ -0,0 +1,94 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { BackupRecoveryPhoneCodeImage } from '../../images';
+import LinkExternal from 'fxa-react/components/LinkExternal';
+import FlowContainer from '../FlowContainer';
+import { GleanClickEventType2FA } from '../../../lib/types';
+import Banner from '../../Banner';
+
+export type FlowSetup2faPromptProps = {
+  localizedPageTitle: string;
+  onContinue: () => void;
+  onCancel: () => void;
+  serviceName: string;
+  localizedErrorMessage?: string;
+};
+
+export const FlowSetup2faPrompt = ({
+  localizedPageTitle,
+  onContinue,
+  onCancel,
+  serviceName,
+  localizedErrorMessage,
+}: FlowSetup2faPromptProps) => {
+  return (
+    <FlowContainer onBackButtonClick={onCancel} title={localizedPageTitle}>
+      {localizedErrorMessage && (
+        <Banner
+          type="error"
+          content={{ localizedHeading: localizedErrorMessage }}
+        />
+      )}
+      <BackupRecoveryPhoneCodeImage ariaHidden />
+      <FtlMsg id="flow-setup-2fa-prompt-heading">
+        <h2 className="font-bold text-xl my-2">
+          Set up two-step authentication
+        </h2>
+      </FtlMsg>
+      <FtlMsg id="flow-setup-2fa-prompt-description" vars={{ serviceName }}>
+        <p className="text-base">
+          {serviceName} requires you set up two-step authentication to keep your
+          account safe.
+        </p>
+      </FtlMsg>
+      <FtlMsg
+        id="flow-setup-2fa-prompt-use-authenticator-apps"
+        elems={{
+          authenticationAppsLink: (
+            <LinkExternal
+              className="link-blue text-sm"
+              href="https://support.mozilla.org/kb/secure-firefox-account-two-step-authentication"
+              gleanDataAttrs={{
+                id: 'two_step_auth_app_link',
+                type: GleanClickEventType2FA.inline,
+              }}
+            >
+              these authenticator apps
+            </LinkExternal>
+          ),
+        }}
+      >
+        <p className="text-sm my-8">
```

Comment:
The smaller text looked odd to me when I first saw the screenshot, but I do see it matches the Figma design. It's a bit unusual to mix different sizing like this, so perhaps check with UX to confirm this is intentional if you share this feeling?